src/components/PhoneVerificationModal.tsx

```diff
@@ -0,0 +1,166 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { useTranslation } from 'react-i18next';
+import { Button, InputField, SelectField } from './forms';
+import { Modal } from './Modal';
+import { Dispatch, FormEvent, SetStateAction, useState } from 'react';
+import { useCountries } from '$app/common/hooks/useCountries';
+import { request } from '$app/common/helpers/request';
+import { endpoint } from '$app/common/helpers';
+import { toast } from '$app/common/helpers/toast/toast';
+import { AxiosError } from 'axios';
+import { ValidationBag } from '$app/common/interfaces/validation-bag';
+import { SmsVerificationModal } from '$app/pages/settings/user/components/SmsVerificationModal';
+import { Alert } from './Alert';
+import { useQueryClient } from 'react-query';
+import { updateCompanyUsers } from '$app/common/stores/slices/company-users';
+import { useDispatch } from 'react-redux';
+import { GenericSingleResourceResponse } from '$app/common/interfaces/generic-api-response';
+import { CompanyUser } from '$app/common/interfaces/company-user';
+
+interface Props {
+  visible: boolean;
+  setVisible: Dispatch<SetStateAction<boolean>>;
+}
+
+export function PhoneVerificationModal(props: Props) {
+  const [t] = useTranslation();
+  const dispatch = useDispatch();
+  const countries = useCountries();
+  const queryClient = useQueryClient();
+
+  const [isSmsModalOpen, setIsSmsModalOpen] = useState<boolean>(false);
+
+  const [countryCode, setCountryCode] = useState('');
+  const [phoneNumber, setPhoneNumber] = useState('');
+
+  const [errors, setErrors] = useState<ValidationBag>();
+
+  const handleClose = () => {
+    props.setVisible(false);
+    setPhoneNumber('');
+    setCountryCode('');
+    setErrors(undefined);
+  };
+
+  const handleSendSMSCode = (event: FormEvent<HTMLButtonElement>) => {
```

Comment:
Two things are wrong here.

1. If you want to prevent the default behavior of the button, you should just use type=button, or as it's in our app `behavior`.
2. You don't wanna do it this way. Please wrap it in the form and then prevent default on the form.